### ISSUE TITLE
Doesnt work?

### ISSUE THREAD

**Issue by bmwdeejay** [func-vehicle/gtablocker#2]
Hello,

My friends are getting the unable to connect message only until i unblock all i suspect its something to do with the way its blocking the ips when i check my firewall rule it shows as something like this from 0.0.0.0 to x.x.x.x

Windows 7 is what im using .. The it deffo blocks just doesnt allow friends for me

**Comment by func-vehicle** [comment]
It should block all but your friend's IPs, so if you have one friend with the IP 5.5.5.5, you should see 0.0.0.0-5.5.5.4 and 5.5.5.6-255.255.255.255 blocked in the scope tab of one of the rules, leaving just 5.5.5.5 open. Not 100% on this, but depending on your setup you may need to port forward in order to be able to host a session with your friends.

**Comment by bmwdeejay** [comment]
thanks for the reply, so you suggest me to port foward on the router yeah? And it would be ports 6672,61455,61456,61457,61458 right?

**Comment by func-vehicle** [comment]
Yep, once those ports are fowarded on UDP it _should_ work.

**Comment by bmwdeejay** [comment]
Yeah sadly that didnt help , Even on second pc can not connect  to the session but as soon as i unblock all then it works no problem.. I have ported fowarded all ports as suggested so im thinking its something to do with my firewall.. The only thing i did there manually was delete the GTA default rules the ones created when the game is installed that wouldnt be it would it?

**Comment by func-vehicle** [comment]
The batch file deletes the default rules, so that would be fine. Currently there are two rules, a 'block' rule and an 'allow' rule. When blocking, allow is disabled and block is enabled, and vice-versa. I'm thinking that the allow rule being disabled is causing even whitelisted connections to be refused on your setup, so I'll try changing it so the allow rule always stays enabled and see if it still works correctly.

**Comment by bmwdeejay** [comment]
I did a little testing 
unblock all = works no problem anyone can join as intended
block all = works no problem no one can join me
block all but friends = no one can join me still
i then did the following 
1) Added my friend and ip into the list
2) Clicked block all but friends 
3) Manually enabled the allow all in the firewall
so i did this test first without adding a friends ip to the list and they were not able to join - i then did the steps above and they where able to join - others still not able to it seems so it would appear its something to do with what you said.

**Comment by func-vehicle** [comment]
I've updated and released version 2.3.0 of the port blocker, which has the fix applied. It should work properly for you now - check out the release at https://github.com/func-vehicle/gtablocker/releases/tag/2.3.0

**Comment by bmwdeejay** [comment]
nice i see it works for that issue but i currently have another issue - i got like 10 players on my friends list but some of them cant join - i have had them double check there ip and send me screenshots showing it and i have the same one on the block allow list but for some reason they can not join me still - sometimes even when i put it on unblock all - have you any idea why this would be ?

**Comment by bmwdeejay** [comment]
So after a little more testing with trying some diff things i found out its something to do with one of the following
windows firewall 
nat type for others 
 because when i block all but friends it allows some of them to join no problem but some of them cant join they get the session not found message however if i unblock all allow them to join and then block all but friends they stay in the session in both cases they are whitelisted and i didnt change there ip so then onto the nat type so originally i had a moderate nat type and after some searching i was able to get my nat type to open but alas it was still not allowing my second pc even when it was connected to a vpn to connect when in block all but friends mode but again when i unblock all and allow them in the session then block but friends they stay in the session

Im not sure what to do next if even there is anything to be done

**Comment by func-vehicle** [comment]
If you have an open NAT type, anybody in theory should be able to join you. It might be that the server is not being advertised properly with the blocker enabled, and that's why your friends get the session not found error. Although if it works for some, really anybody should be able to join. It definitely sounds like it's working once they're in though, normally when you start blocking in an existing session you'll either be kicked or everyone would freeze, so your friends being unaffected is a good sign.

As for the second computer not being able to join, I know having two computers / consoles on the same network trying to join each other can be troublesome as some games don't like that. With the VPN disabled, if it doesn't work with the blocker disabled, then it's probably the game's fault. If you have the VPN enabled, it should work with the VPN's IP whitelisted, not sure why it wouldn't.

**Comment by func-vehicle** [comment]
I just realized that the rules are blocking all local connections - I'll fix this and put out an update to setup.bat soon.

**Comment by func-vehicle** [comment]
Updated setup.bat is available - https://github.com/func-vehicle/gtablocker/blob/master/setup.bat

**Comment by bmwdeejay** [comment]
I downloaded and ran the setup bat as suggested, have been busy irl the last couple days this is the first chace i get to reply but i recorded a video of another issue that is more of a problem because it seems for me now its not working as expected at all.. see the video to see what i mean > https://youtu.be/lZcGn5lgqaI

**Comment by func-vehicle** [comment]
I've done some further research and realized that the change I made does not do what I thought it did. Instead it looks like I'll have to update the application to automatically add an exception from 192.168.0.0–192.168.255.255. I've reverted setup.bat for now.

**Comment by bmwdeejay** [comment]
ok i still can not kick everyone out using the block all but friends and when i use block all it doesnt work no more - maybe rockstar changed something? either way the program doesnt work at any anymore for me

**Comment by func-vehicle** [comment]
Still works for me - replace your .bat and .exe with the latest version (https://github.com/func-vehicle/gtablocker/releases/tag/2.4.0), rerun the .bat after changing the GTA5.exe location, then try blocking in the app again.

**Comment by Tampa** [comment]
I used to just block via firewall, Kaspersky in this case, but that seemed to stop working as well. What does still work like a charm is using openvpn to connect to a server that has port 6672 blocked, but overrules for local addresses and my own IP. Now I only tested with another computer on local NAT, but technically the server can overrule any other IP as well and thus should be able to create a one-way session only those in iptables exclusions are allowed in. Connection to the openvpn also seems to empty out the session I am in, but leaves anyone on the local network in it, though for some reason that does make them more prone to game crashes. Maybe that's a hint, not sure.

**Comment by func-vehicle** [comment]
Last time I tried using this it didn't work either, think an update did something to this method (intentionally or not).
